### ISSUE TITLE
Some Errors

### ISSUE THREAD

**Issue by TeQTabanca** [MyPetORG/MyPet#1537]
Plugin Version: MyPet-3.12-SNAPSHOT-B1682
Minecraft Version: 1.12.2 SPİGOT

    "Ride": {
      "Upgrades": {
        "1": {
          "Speed": "+100",
          "Active": true,
          "JumpHeight": "+10.25"

This is how I set it up. but while jumping the pet dies and respawns but it still looks like I'm riding the pet and the console shows an error. You can understand more easily with the video.

https://user-images.githubusercontent.com/89253304/160915428-d9955d09-4c4e-4e30-8991-bc879f9ee496.mp4

```
[20:49:54 ERROR]: Could not pass event EntityDeathEvent to MyPet v3.12
org.bukkit.event.EventException: null
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[server.jar:git-Spigot-79a30d7-acbc348]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[server.jar:git-Spigot-79a30d7-acbc348]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:500) [server.jar:git-Spigot-79a30d7-acbc348]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:485) [server.jar:git-Spigot-79a30d7-acbc348]
        at org.bukkit.craftbukkit.v1_12_R1.event.CraftEventFactory.callEntityDeathEvent(CraftEventFactory.java:395) [server.jar:git-Spigot-79a30d7-acbc348]
        at org.bukkit.craftbukkit.v1_12_R1.event.CraftEventFactory.callEntityDeathEvent(CraftEventFactory.java:388) [server.jar:git-Spigot-79a30d7-acbc348]
        at net.minecraft.server.v1_12_R1.EntityLiving.die(EntityLiving.java:1119) [server.jar:git-Spigot-79a30d7-acbc348]
        at de.Keyle.MyPet.compat.v1_12_R1.entity.EntityMyPet.die(EntityMyPet.java:649) [MyPet-3.12-SNAPSHOT-B1682.jar:?]
        at net.minecraft.server.v1_12_R1.EntityLiving.damageEntity(EntityLiving.java:956) [server.jar:git-Spigot-79a30d7-acbc348]
        at net.minecraft.server.v1_12_R1.EntityLiving.e(EntityLiving.java:1211) [server.jar:git-Spigot-79a30d7-acbc348]
        at de.Keyle.MyPet.compat.v1_12_R1.entity.EntityMyPet.e(EntityMyPet.java:1100) [MyPet-3.12-SNAPSHOT-B1682.jar:?]
        at net.minecraft.server.v1_12_R1.Block.fallOn(Block.java:533) [server.jar:git-Spigot-79a30d7-acbc348]
        at net.minecraft.server.v1_12_R1.Entity.a(Entity.java:1034) [server.jar:git-Spigot-79a30d7-acbc348]
        at net.minecraft.server.v1_12_R1.EntityLiving.a(EntityLiving.java:195) [server.jar:git-Spigot-79a30d7-acbc348]
        at net.minecraft.server.v1_12_R1.Entity.move(Entity.java:815) [server.jar:git-Spigot-79a30d7-acbc348]
        at de.Keyle.MyPet.compat.v1_12_R1.entity.EntityMyPet.ride(EntityMyPet.java:772) [MyPet-3.12-SNAPSHOT-B1682.jar:?]
        at de.Keyle.MyPet.compat.v1_12_R1.entity.EntityMyPet.a(EntityMyPet.java:1171) [MyPet-3.12-SNAPSHOT-B1682.jar:?]
        at de.Keyle.MyPet.compat.v1_12_R1.entity.EntityMyPet.n(EntityMyPet.java:975) [MyPet-3.12-SNAPSHOT-B1682.jar:?]
        at net.minecraft.server.v1_12_R1.EntityLiving.B_(EntityLiving.java:1946) [server.jar:git-Spigot-79a30d7-acbc348]
        at net.minecraft.server.v1_12_R1.EntityInsentient.B_(EntityInsentient.java:246) [server.jar:git-Spigot-79a30d7-acbc348]
        at de.Keyle.MyPet.compat.v1_12_R1.entity.EntityMyPet.B_(EntityMyPet.java:1254) [MyPet-3.12-SNAPSHOT-B1682.jar:?]
        at net.minecraft.server.v1_12_R1.World.entityJoinedWorld(World.java:1631) [server.jar:git-Spigot-79a30d7-acbc348]
        at net.minecraft.server.v1_12_R1.World.h(World.java:1601) [server.jar:git-Spigot-79a30d7-acbc348]
        at net.minecraft.server.v1_12_R1.World.tickEntities(World.java:1435) [server.jar:git-Spigot-79a30d7-acbc348]
        at net.minecraft.server.v1_12_R1.WorldServer.tickEntities(WorldServer.java:668) [server.jar:git-Spigot-79a30d7-acbc348]
        at net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:817) [server.jar:git-Spigot-79a30d7-acbc348]
        at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:406) [server.jar:git-Spigot-79a30d7-acbc348]
        at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:679) [server.jar:git-Spigot-79a30d7-acbc348]
        at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:577) [server.jar:git-Spigot-79a30d7-acbc348]
        at java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: java.lang.NoClassDefFoundError: org/bukkit/GameRule
        at de.Keyle.MyPet.listeners.MyPetEntityListener.sendDeathMessage(MyPetEntityListener.java:569) ~[MyPet-3.12-SNAPSHOT-B1682.jar:?]
        at de.Keyle.MyPet.listeners.MyPetEntityListener.onMyPet(MyPetEntityListener.java:551) ~[MyPet-3.12-SNAPSHOT-B1682.jar:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[server.jar:git-Spigot-79a30d7-acbc348]
        ... 29 more
Caused by: java.lang.ClassNotFoundException: org.bukkit.GameRule
        at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:105) ~[server.jar:git-Spigot-79a30d7-acbc348]
        at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:100) ~[server.jar:git-Spigot-79a30d7-acbc348]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:589) ~[?:?]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:522) ~[?:?]
        at de.Keyle.MyPet.listeners.MyPetEntityListener.sendDeathMessage(MyPetEntityListener.java:569) ~[MyPet-3.12-SNAPSHOT-B1682.jar:?]
        at de.Keyle.MyPet.listeners.MyPetEntityListener.onMyPet(MyPetEntityListener.java:551) ~[MyPet-3.12-SNAPSHOT-B1682.jar:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[server.jar:git-Spigot-79a30d7-acbc348]
        ... 29 more
```

        "1": {
          "CanFly": true
        }
        
This feature is not working

https://user-images.githubusercontent.com/89253304/160916709-a47ba180-c4b7-418c-9829-f8f0f571351d.mp4

**Comment by Jakllp** [comment]
I'll be checking out these problems soon

**Comment by Jakllp** [comment]
The error (and actually another bug with horse-pets dying whilst they are being ridden) are fixed on my end and will be pushed with some other updates soon

**Comment by TeQTabanca** [comment]
> About the CanFly:
> 
> After reading your issue it looks like you set up your skilltree's Ride-Skill like this:
> 
> ```
>     "Ride": {
>       "Upgrades": {
>         "1": {
>           "Speed": "+100",
>           "Active": true,
>           "JumpHeight": "+10.25"
>         },
>         "1": {
>           "CanFly": true
>         }
>       }
>     }
> ```
> 
> Which will not work as now a skilltree has 2 upgrade-sections for level 1
> 
> It would need to look like this:
> 
> ```
>     "Ride": {
>       "Upgrades": {
>         "1": {
>           "Speed": "+100",
>           "Active": true,
>           "CanFly": true,
>           "JumpHeight": "+10.25"
>         }
>       }
>     }
> ```
> 
> My suggestion: Just use the [SkilltreeCreator](https://wiki.mypet-plugin.de/systems/skilltrees/skilltreecreator) (after fixing this in the file itself)

**i tried this but still getting the same result**

**Comment by TeQTabanca** [comment]
[Ride.st.json.zip](https://github.com/MyPetORG/MyPet/files/8418836/Ride.st.json.zip)

**Comment by TeQTabanca** [comment]
I tried it now and it works. The reason is that I didn't get a strap on my hand. 😌

**Comment by Jakllp** [comment]
I tried your skilltree with Spigot 1.12.2 and it worked as expected... Don't really know what the issue is atm - you might want to try `/petadmin reload skilltrees`

**Comment by TeQTabanca** [comment]
**By the way, even though I deleted all the files in the "Mypet" file, the plugin does not give an error, but the files are not restored.**

**Comment by Jakllp** [comment]
It isn't actually supposed to replace missing skilltrees as you wouldn't want it to just always paste the default skilltrees in there after you created your own ones.  
It only does it when the entire MyPet directory is missing

**Comment by Jakllp** [comment]
Well if you delete the entire folder you'll need to restart the entire server

**Comment by Jakllp** [comment]
Yes, that is the expected behavior.
The command only reads the config files and checks for changes but will not create new files.

So if you adjust a skilltree or a shop (for example) you don't have to reboot the server for your changes to be enabled

**Comment by Jakllp** [comment]
Fixed with `3.12-SNAPSHOT-B1685`
By https://github.com/MyPetORG/MyPet/commit/3bd9a91dc4a7f120a3387c9236d76f99ff8ae244
